providers/tailscale/resources/discovery.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package resources
+
+import (
+	"go.mondoo.com/cnquery/v11/llx"
+	"go.mondoo.com/cnquery/v11/providers-sdk/v1/inventory"
+	"go.mondoo.com/cnquery/v11/providers-sdk/v1/plugin"
+	"go.mondoo.com/cnquery/v11/providers/tailscale/connection"
+	"go.mondoo.com/cnquery/v11/utils/stringx"
+)
+
+func Discover(runtime *plugin.Runtime, opts map[string]string) (*inventory.Inventory, error) {
+	conn := runtime.Connection.(*connection.TailscaleConnection)
+
+	in := &inventory.Inventory{Spec: &inventory.InventorySpec{
+		Assets: []*inventory.Asset{},
+	}}
+
+	targets := handleTargets(conn.Asset().Connections[0].Discover.Targets)
+	list, err := discover(runtime, targets)
+	if err != nil {
+		return in, err
+	}
+
+	in.Spec.Assets = list
+	return in, nil
+}
+
+func handleTargets(targets []string) []string {
+	if stringx.ContainsAnyOf(targets, connection.DiscoveryAll, connection.DiscoveryAuto) {
+		return []string{
+			connection.DiscoveryDevices,
+		}
+	}
+	return targets
+}
+
+func discover(runtime *plugin.Runtime, targets []string) ([]*inventory.Asset, error) {
+	conn := runtime.Connection.(*connection.TailscaleConnection)
+	conf := conn.Asset().Connections[0]
+	assetList := []*inventory.Asset{}
+
+	cf, err := getMqlTailscale(runtime)
```

Comment:
I think we want also discover the root asset (org/tailnet).

Reply:
I don't think we can fetch this information.